instrumentation/apache-log4j-2/src/main/java/com/nr/agent/instrumentation/log4j2/AgentUtil.java

```diff
@@ -52,11 +56,15 @@ public static void recordNewRelicLogEvent(LogEvent event) {
                 String formattedMessage = message.getFormattedMessage();
                 // Bail out and don't create a LogEvent if log message is empty
                 if (formattedMessage != null && !formattedMessage.isEmpty()) {
-                    HashMap<String, Object> logEventMap = new HashMap<>(DEFAULT_NUM_OF_LOG_EVENT_ATTRIBUTES);
+                    HashMap<String, Object> logEventMap = new HashMap<>(getDefaultNumOfLogEventAttributes(event.getContextData()));
 
                     logEventMap.put(MESSAGE, formattedMessage);
                     logEventMap.put(TIMESTAMP, event.getTimeMillis());
 
+                    if (isApplicationLoggingForwardingIncludeMdcEnabled() && event.getContextData() != null) {
+                        event.getContextData().forEach((key, value) -> logEventMap.put(MDC_ATTRIBUTE_PREFIX + key, value));
```

Comment:
I'd love to make the prefix optional.

When working with distributed microservices that are multi-language / framework its nice to have a single context key to search for.

I wouldn't want to have to write queries that are `WHERE mdc.foo = 'bar' AND foo = 'bar'` if I could just write `WHERE foo = 'bar'`

Reply:
I completely agree with you, I'll incorporate it.